ihmc-perception/src/main/java/us/ihmc/sensors/RealSensePublisher.java

```diff
@@ -0,0 +1,77 @@
+package us.ihmc.sensors;
+
+import perception_msgs.msg.dds.ImageMessage;
+import us.ihmc.commons.thread.ThreadTools;
+import us.ihmc.communication.CommunicationMode;
+import us.ihmc.communication.PerceptionAPI;
+import us.ihmc.communication.ROS2Tools;
+import us.ihmc.communication.ros2.ROS2DemandGraphNode;
+import us.ihmc.communication.ros2.ROS2Helper;
+import us.ihmc.euclid.referenceFrame.ReferenceFrame;
+import us.ihmc.perception.RawImage;
+import us.ihmc.perception.realsense.RealsenseConfiguration;
+import us.ihmc.perception.realsense.RealsenseDeviceManager;
+import us.ihmc.ros2.ROS2Node;
+import us.ihmc.ros2.ROS2Topic;
+import us.ihmc.tools.thread.ExecutorServiceTools;
+import us.ihmc.tools.thread.RestartableThread;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Supplier;
+
+public class RealSensePublisher
+{
+   private static final String REALSENSE_SERIAL_NUMBER = System.getProperty("d455.serial.number", "213522252883");
+   private static final ROS2Topic<ImageMessage> REALSENSE_COLOR_TOPIC = PerceptionAPI.D455_COLOR_IMAGE;
+   private static final ROS2Topic<ImageMessage> REALSENSE_DEPTH_TOPIC = PerceptionAPI.D455_DEPTH_IMAGE;
+
+   private final RealsenseColorDepthImageRetriever realsenseImageRetriever;
+   private final RealsenseColorDepthImagePublisher realsenseImagePublisher;
+
+   private final ROS2DemandGraphNode realsenseDemandNode;
+
+   protected final ScheduledExecutorService executorService = ExecutorServiceTools.newScheduledThreadPool(1,
+                                                                                                          getClass(),
+                                                                                                          ExecutorServiceTools.ExceptionHandling.CATCH_AND_REPORT);
+
+   public RealSensePublisher(Supplier<ReferenceFrame> realsenseFrameSupplier, ROS2Helper ros2Helper)
+   {
+      realsenseDemandNode = new ROS2DemandGraphNode(ros2Helper, PerceptionAPI.REQUEST_REALSENSE_POINT_CLOUD);
+
+      realsenseImageRetriever = new RealsenseColorDepthImageRetriever(new RealsenseDeviceManager(),
+                                                                      REALSENSE_SERIAL_NUMBER,
+                                                                      RealsenseConfiguration.D455_COLOR_720P_DEPTH_720P_30HZ,
+                                                                      realsenseFrameSupplier,
+                                                                      realsenseDemandNode);
+
+      realsenseImagePublisher = new RealsenseColorDepthImagePublisher(REALSENSE_DEPTH_TOPIC, REALSENSE_COLOR_TOPIC);
+
+      RestartableThread realsenseProcessAndPublishThread = new RestartableThread("RealsenseProcessAndPublish", this::processAndPublishRealsense);
+      realsenseProcessAndPublishThread.start();
```

Comment:
I don't think you need to start a separate thread here. Could just do a `while (true)` thing if you want.